File: 2022/June /Day 1/RunningSum.cpp
// Source : https://leetcode.com/problems/running-sum-of-1d-array/submissions/
class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        vector<int> out;
        int rs=0;
        for(auto&i:nums)
        {
            rs+=i;
            out.insert(out.end(),rs);
        }
        return out;
    }
};

File: 2024/Jun/runningSum.go
func runningSum(nums []int) []int {

	s := make([]int, 0, len(nums))
	// For loop in Go
	for i := 0; i < len(nums); i++ {
		if i > 0 {
			s = append(s, s[i-1]+nums[i])
		} else {
			s = append(s, nums[i])

		}
	}
	return s
}